### PR TITLE
RATIS-2178. Add a log message whenever deleting a RaftLog segment.

### PR DESCRIPTION
**Pull request by szetszwo**
See RATIS-2178

### CODE REVIEW

File: ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogCache.java [hunk 1]
```diff
@@ -154,7 +159,8 @@ long maxEndIndex() {
 
     @Override
     public String toString() {
-      return "toTruncate: " + toTruncate
+      return reason
+          + "\ntoTruncate: " + toTruncate
```
Comment (OneSizeFitsQuorum) [inline ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogCache.java:163]: \n toTruncate:

File: ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLog.java [hunk 2]
```diff
@@ -405,12 +405,13 @@ protected CompletableFuture<Long> purgeImpl(long index) {
...
       SegmentedRaftLogCache.TruncationSegments ts = cache.purge(index);
       updateSnapshotIndexFromStateMachine();
-      LOG.debug("purging segments:{}", ts);
       if (ts != null) {
+        LOG.info("{}: {}", getName(), ts);
         Task task = fileLogWorker.purge(ts);
         return task.getFuture();
       }
     }
+    LOG.info("{}: purge({}) found nothing to purge.", getName(), index);
```
Comment (OneSizeFitsQuorum) [inline ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLog.java:414]: It seems this line will cause a lot of info log? If we don't actually delete the file here, maybe we don't need to log here

**Review by OneSizeFitsQuorum — COMMENTED** [review]
Awesome work!
Looking forward to find potential bugs here after this change

File: ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLog.java (hunk 2 above)
Comment (szetszwo) [reply ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLog.java:414]: Sure, let's change it to debug.

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Comment by szetszwo** [comment]
@OneSizeFitsQuorum , thanks a lot for reviewing this!

**Pull request merged**